gazebo/rendering/Visual_TEST.cc

```diff
@@ -1588,6 +1588,33 @@ TEST_F(Visual_TEST, ConvertVisualType)
       rendering::Visual::ConvertVisualType(msgs::Visual::PHYSICS));
 }
 
+TEST_F(Visual_TEST, collisionZero)
+{
+  // This test checks that there isn't a segmentation fault when inserting zero collision geometries
+  // Load a world containing 3 simple shapes with collision geometry equals to zero
+  Load("worlds/collision_zero.world");
+
+  // Get the scene
+  gazebo::rendering::ScenePtr scene = gazebo::rendering::get_scene();
+  ASSERT_NE(scene, nullptr);
+
+  // Wait until all models are inserted
+  int sleep = 0;
+  int maxSleep = 10;
+  rendering::VisualPtr box, sphere, cylinder;
+  while ((!box || !sphere || !cylinder) && sleep < maxSleep)
+  {
+    event::Events::preRender();
+    event::Events::render();
+    event::Events::postRender();
+
+    box = scene->GetVisual("box");
+    cylinder = scene->GetVisual("cylinder");
+    sphere = scene->GetVisual("sphere");
```

Comment:
Should you check that these aren't `nullptr` after the loop?

Reply:
Test is passing:

```
133: [ RUN      ] Visual_TEST.CollisionZero
133: [Msg] Waiting for master.
133: [Msg] Connected to gazebo master @ http://127.0.0.1:11345
133: [Msg] Publicized address: 172.17.0.2
133: [Dbg] [ServerFixture.cc:203] ServerFixture load in 0.6 seconds, timeout after 600 seconds
133: [Err] [Visual.cc:823] box::link::collision__COLLISION_VISUAL__: Size of the collision contains one or several zeros. Collisions may not visualize properly.
133: [Err] [Visual.cc:823] sphere::link::collision__COLLISION_VISUAL__: Size of the collision contains one or several zeros. Collisions may not visualize properly.
133: [Err] [Visual.cc:823] cylinder::link::collision__COLLISION_VISUAL__: Size of the collision contains one or several zeros. Collisions may not visualize properly.
133: [Dbg] [ServerFixture.cc:129] ServerFixture::Unload
133: [       OK ] Visual_TEST.CollisionZero (2810 ms)
133: [ RUN      ] Visual_TEST.Scale
```